examples/domainmodel/src/cli/cli-util.ts

```diff
@@ -0,0 +1,48 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import fs from 'fs';
+import colors from 'colors';
+import { AstNode, DefaultDocumentValidator, LangiumDocument, LangiumDocumentConfiguration, LangiumServices } from 'langium';
+
+export function extractDocument(fileName: string, languageId: string, extensions: string[], services: LangiumServices): LangiumDocument {
+    if (!new RegExp(`^.*${extensions?.join('|')}$`).test(fileName)) {
```

Comment:
I don't think this regex will end up working as expected. Assuming we have two file extensions `.a` and `.b`, the regex will end up looking like this: `/^.*.a|.b$/`. This contains several issues:

* Every file extension starting with `.` will match any character and therefore should be escaped.
* The alternative will split this regex into `.*.a` and `.b` which isn't optimal, as the `.b` extension will probably never match the file name.

I wouldn't recommend doing this with regex but instead use the `path` import:

```ts
import path from 'path';
...
if (extensions.includes(path.extname(fileName)) { ... }
```